packages/browser/src/node/commands/screenshotMatcher/utils.ts

```diff
@@ -0,0 +1,281 @@
+import type { BrowserCommandContext, BrowserConfigOptions } from 'vitest/node'
+import type { ScreenshotMatcherOptions } from '../../../../context'
+import type { AnyCodec } from './codecs'
+import { platform } from 'node:os'
+import { basename, dirname, extname, join, relative, resolve, sep } from 'node:path'
+import { deepMerge } from '@vitest/utils'
+import { PlaywrightBrowserProvider } from '../../providers/playwright'
+import { WebdriverBrowserProvider } from '../../providers/webdriver'
+import { takeScreenshot } from '../screenshot'
+import { getCodec } from './codecs'
+import { getComparator } from './comparators'
+
+type GlobalOptions = Required<
+  NonNullable<
+    NonNullable<BrowserConfigOptions['expect']>['toMatchScreenshot']
+  >
+>
+
+const defaultOptions = {
+  comparatorName: 'pixelmatch',
+  // these are handled by each comparator on its own
+  comparatorOptions: {},
+  screenshotOptions: {
+    animations: 'disabled',
+    caret: 'hide',
+    fullPage: false,
+    maskColor: '#ff00ff',
+    omitBackground: false,
+    scale: 'css',
+  },
+  timeout: 5_000,
+  resolveDiffPath: ({
+    arg,
+    ext,
+    root,
+    browserName,
+    testFileDirectory,
+    testFileName,
+  }) => resolve(
+    root,
+    '__diffs__',
+    testFileDirectory,
+    testFileName,
+    `${arg}-${browserName}${ext}`,
+  ),
+  resolveScreenshotPath: ({
+    arg,
+    ext,
+    root,
+    screenshotDirectory,
+    testFileDirectory,
+    testFileName,
+    browserName,
+  }) => resolve(
+    root,
+    testFileDirectory,
+    screenshotDirectory,
+    testFileName,
+    `${arg}-${browserName}${ext}`,
+  ),
+} satisfies GlobalOptions
+
+type SupportedCodecs = Parameters<typeof getCodec>[0]
+
+const supportedExtensions = ['png'] satisfies SupportedCodecs[]
+
+export function resolveOptions(
+  {
+    context,
+    name,
+    options,
+    testName,
+  }: {
+    context: BrowserCommandContext
+    name: string
+    testName: string
+    options: ScreenshotMatcherOptions
+  },
+): {
+    codec: ReturnType<typeof getCodec>
+    comparator: ReturnType<typeof getComparator>
+    resolvedOptions: GlobalOptions
+    paths: {
+      reference: string
+      diffs: {
+        reference: string
+        actual: string
+        diff: string
+      }
+    }
+  } {
+  if (context.testPath === undefined) {
+    throw new Error('`resolveOptions` has to be used in a test file')
+  }
+
+  const resolvedOptions = deepMerge<GlobalOptions>(
+    Object.create(null),
+    defaultOptions,
+    context.project.config.browser.expect?.toMatchScreenshot ?? {},
```

Comment:
I don't think these are actually read, but will work on it later, I mostly wanted to move all the defaults and stuff in one place instead of having them spread across the project.